crosschain/src/main/java/org/hyperledger/besu/crosschain/core/CrosschainProcessor.java

```diff
@@ -131,12 +138,69 @@ boolean processSubordinates(
-        // view.
-        LOG.info("Crosschain Result: " + result.toString());
-        subordinateTransactionsAndView.addSignedResult(result);
+        if ((!processSubordinateTransactions
+            && subordinateTransactionsAndView.getType().isSubordinateView())) {
+
+          // Decode the response
+          SubordinateViewResultMessage viewResultMessage =
+              (SubordinateViewResultMessage) ThresholdSignedMessage.decodeEncodedMessage(result);
+
+          // Obtain the blockchain public key from the coordination contract using the sidechainId
+          Optional<BigInteger> coordChainId = transaction.getCrosschainCoordinationBlockchainId();
+          Optional<Address> coordAddr = transaction.getCrosschainCoordinationContractAddress();
+          if (coordChainId.isEmpty() || coordAddr.isEmpty()) {
+            LOG.error("Coordination Chain is not set up");
+            return true;
+          }
+          String coordIpAddrAndPort =
+              coordContractManager.getIpAndPort(coordChainId.get(), coordAddr.get());
+          BigInteger publicKey =
+              new OutwardBoundConnectionManager(this.nodeKeys)
+                  .getPublicKeyFromCoordContract(
+                      coordIpAddrAndPort,
+                      coordChainId.get(),
+                      coordAddr.get(),
+                      sidechainId,
+                      viewResultMessage.getKeyVersion());
+          LOG.info(
+              "Obtained the public key {} from crosschain coordination contract.",
+              publicKey.toString(16));
+
+          // Verify the signature
+          BlsPoint publicKeyBlsPoint = BlsPoint.load(publicKey.toByteArray());
+
+          BlsCryptoProvider cryptoProvider =
+              BlsCryptoProvider.getInstance(
+                  BlsCryptoProvider.CryptoProviderTypes.LOCAL_ALT_BN_128,
+                  BlsCryptoProvider.DigestAlgorithm.KECCAK256);
+
+          boolean signatureVerification =
+              cryptoProvider.verify(
+                  publicKeyBlsPoint,
+                  viewResultMessage.getEncodedCoreMessage().extractArray(),
+                  BlsPoint.load(viewResultMessage.getSignature().getByteArray()));
+
+          if (signatureVerification) {
+            LOG.info("The signature of Subordinate View Result message verified.");
+          } else {
+            LOG.error("Verification of the subordinate view result message's signature failed.");
+            return true;
+          }
+
+          // Check that the Subordiante View hash returned matches the submitted subordiante view.
+          if (subordinateTransactionsAndView.hash().equals(viewResultMessage.getTxHash())) {
+            LOG.info("The hash of the subordinate view result transaction matches.");
+          } else {
+            LOG.error("The hash of the subordinate view result transaction does not match");
+            return true;
```

Comment:
This is a bit of an awkward method name / return value. Maybe it'd be good to change it so it's clearer what it does?

Reply:
(I mean - this method sounds like it could be named `didSubodinateFail()` or some such)

Reply:
didSubordinateFail() does not capture "processing" of subordinates! Any alternate suggestions? @drinkcoffee